lib/cn/cn_tree.c

```diff
@@ -1965,20 +1961,263 @@ static merr_t
 cn_subspill_apply(struct subspill *ss)
 {
     struct cn_compaction_work *w = ss->ss_work;
-    merr_t err;
+    merr_t err = 0;
+    bool delete_node = false;
+    struct cn_tree_node *outnode;
+
+    if (!ss->ss_added)
+        return 0;
+
+    if (ss->ss_is_zspill) {
+        err = cn_move(w, w->cw_node, ss->ss_zspill.zsp_src_list,
+                      w->cw_kvset_cnt, delete_node, ss->ss_node);
+
+        w->cw_output_nodev = &outnode;
+
+    } else {
+        err = cn_subspill_commit(ss);
+    }
 
-    err = cn_subspill_commit(ss);
     if (err)
         return err;
 
-    if (ss->ss_added) {
-        w->cw_output_nodev[0] = ss->ss_node;
-        w->cw_checkpoint(w);
-    }
+    w->cw_output_nodev[0] = ss->ss_node;
+    w->cw_checkpoint(w);
 
     return 0;
 }
 
+/* Tree must be locked when this is called */
+struct cn_tree_node *
+cn_kvset_can_zspill(struct kvset *ks, struct route_map *map)
+{
+    unsigned char maxkbuf[HSE_KVS_KEY_LEN_MAX];
+    const void *maxkey, *maxptkey, *minkey;
+    uint16_t maxklen, maxptklen, minklen;
+    struct route_node *minnode;
+
+    maxklen = minklen = maxptklen = 0;
+    maxptkey = NULL;
```

Comment:
initialization seems unnecessary since the calls to get min/max keys will initialize them.

Reply:
This is just to appease clang-tidy.

Reply:
Screw clang-tidy if it can't figure that out.  @tristan957, is clang-tidy that weak?

Reply:
Perhaps it's because they're in a different file and cannot be seen from here?

Reply:
It wasn't a clang-tidy issue. It is a clang issue, and I am not sure why. Somehow it is related to the MTF code path

Reply:
https://github.com/hse-project/hse/actions/runs/3161465465/jobs/5147080695#step:7:708

Reply:
Ok, so analysis of the issue:

Decl of `kvset_minkey()`:

```c
void
kvset_maxkey(struct kvset *ks, const void **maxkey, uint16_t *maxklen)
```

Definition of the mock:

```c
#ifndef MTF_MOCK_IMPL_kvset
#define kvset_minkey(...) \
({ \
	if (!mapi_enabled || !mapi_inject_check(mapi_idx_kvset_minkey, 0)) \
		(mtfm_kvset_kvset_minkey_get())(__VA_ARGS__); \
	(void)0; \
})
#endif /* MTF_MOCK_IMPL_kvset */
```

What clang is warning about is that in the event of the `if` statement within the mock failing, the out parameters of `kvset_minkey()` are uninitialized, which is an entirely correct warning to throw in my opinion.

I see three solutions. Pacify clang as Guarav has done, turn `-Wsometimes-uninitialized` off, or generate pragmas that turn the warning off in all mock macro definitions. Happy to entertain any of the ideas.

Reply:
Sigh, this happens all over due to our we interpose our mocks, e.g., happens to hse_meminfo() all over the place...

Reply:
We must not disable the warning, otherwise the mocking interposer will leave the var undefined if the mocked function isn't called...

Reply:
No good answer it seems.